src/Umbraco.Cms.ManagementApi/Controllers/Document/Tree/DocumentTreeItemsController.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Core.Cache;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Security;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.ManagementApi.Services.Entities;
+using Umbraco.Cms.ManagementApi.ViewModels.Tree;
+
+namespace Umbraco.Cms.ManagementApi.Controllers.Document.Tree;
+
+public class DocumentTreeItemsController : DocumentTreeControllerBase
+{
+    public DocumentTreeItemsController(
+        IEntityService entityService,
+        IUserStartNodeEntitiesService userStartNodeEntitiesService,
+        IPublicAccessService publicAccessService,
+        AppCaches appCaches,
+        IBackOfficeSecurityAccessor backofficeSecurityAccessor)
+        : base(entityService, userStartNodeEntitiesService, publicAccessService, appCaches, backofficeSecurityAccessor)
+    {
+    }
+
+    [HttpGet("items")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(PagedResult<DocumentTreeItemViewModel>), StatusCodes.Status200OK)]
+    public async Task<ActionResult<PagedResult<DocumentTreeItemViewModel>>> Items([FromQuery(Name = "key")] Guid[] keys)
```

Comment:
Not sure it's a great idea to pass the keys as query strings, since it will get too long if you request a lot of items

Reply:
The decision for now is to let the client fetch in batches to stay within the URL limitatation